course5/homework5/vector.cc

```diff
@@ -0,0 +1,131 @@
+#include "vector.h"
+#include <iostream>
+#include <cmath>
+
+#define Pi 3.14159265
+
+namespace vector{
+
+    Vector2D::Vector2D(double x, double y) : x(x), y(y) {}
+    Vector2D::Vector2D(const Vector2D& src) : x(src.x), y(src.y)  {}
+    Vector2D& Vector2D::operator=(const Vector2D& src)
+    { 
+        x = src.x;
+        y = src.y;
+        return *this;
+    }
+
+    double Vector2D::GetX() const { return x;}
+    double Vector2D::GetY() const { return y;}
+
+    void Vector2D::SetX(double x)
+    {
+        this->x = x;
+    }
+
+    void Vector2D::SetY(double y)
+    {
+        this->y = y;
+    }
+
+    double Vector2D::GetR() const { return sqrt(x*x + y*y); }
```

Comment:
You must have missed this part of the lecture:
```sqrt``` is a C function that compiles and works here sort of by accident. But C++ only guarantees ```std::sqrt```. There may be a C+ compiler now, or in the future, that will fail to compile the code.